src/common/sys_util.h

```diff
@@ -105,6 +105,9 @@ util_mutex_unlock(pthread_mutex_t *m)
  * caller perspective. If pthread_rwlock_unlock failed, this function aborts
  * the program.
  */
+#ifdef _WIN32
```

Comment:
Why is this the only function that "requires lock held"? All of the unlock functions in this file have this requirement. Also, shouldn't this be located in the pthread_rwlock_unlock function instead?